src/main.rs

```diff
@@ -477,10 +504,10 @@ impl Decoder {
             Format::EndOfInput => Ok(Decoder::EndOfInput),
             Format::Byte(bs) => Ok(Decoder::Byte(*bs)),
             Format::Alt(a, b) => {
-                let da = Box::new(Decoder::compile(a, next)?);
-                let db = Box::new(Decoder::compile(b, next)?);
-                if let Some(tree) = MatchTree::build(&[a, b], next) {
-                    Ok(Decoder::If(Cond::MatchTree(tree), da, db))
+                let da = Decoder::compile(a, next)?;
+                let db = Decoder::compile(b, next)?;
+                if let Some(switch) = MatchTree::build(&[a, b], next) {
+                    Ok(Decoder::Branch(switch, vec![da, db]))
```

Comment:
Use `tree` instead of `switch` here I think.

Reply:
Oops, missed this!

Reply:
Pushed a fix to `main`